builtin/credential/cert/path_crls.go

```diff
@@ -0,0 +1,240 @@
+package cert
+
+import (
+	"crypto/x509"
+	"fmt"
+	"math/big"
+	"strings"
+	"sync"
+
+	"github.com/fatih/structs"
+	"github.com/hashicorp/vault/helper/certutil"
+	"github.com/hashicorp/vault/logical"
+	"github.com/hashicorp/vault/logical/framework"
+)
+
+func pathCRLs(b *backend) *framework.Path {
+	return &framework.Path{
+		Pattern: "crls/" + framework.GenericNameRegex("name"),
+		Fields: map[string]*framework.FieldSchema{
+			"name": &framework.FieldSchema{
+				Type:        framework.TypeString,
+				Description: "The name of the certificate",
+			},
+
+			"crl": &framework.FieldSchema{
+				Type: framework.TypeString,
+				Description: `The public certificate that should be trusted.
+May be DER or PEM encoded. Note: the expiration time
+is ignored; if the CRL is no longer valid, delete it
+using the same name as specified here.`,
+			},
+		},
+
+		Callbacks: map[logical.Operation]framework.OperationFunc{
+			logical.DeleteOperation: b.pathCRLDelete,
+			logical.ReadOperation:   b.pathCRLRead,
+			logical.WriteOperation:  b.pathCRLWrite,
+		},
+
+		HelpSynopsis:    pathCRLsHelpSyn,
+		HelpDescription: pathCRLsHelpDesc,
+	}
+}
+
+var (
+	crls           = map[string]CRLInfo{}
+	crlUpdateMutex = &sync.RWMutex{}
```

Comment:
This should be on the backend and not global, since otherwise it causes cross-backend synchronization